src/rtree.ml

```diff
@@ -178,6 +178,15 @@ module Make (E : Envelope) (V : Value with type envelope = E.t) = struct
       0 partitions lst
     |> List.map List.rev
     |> List.filter_map (fun v -> if v = [] then None else Some v)
+  
+  (* Rtree Iter Function *)
+  let rec iter rtree f =
+    match rtree with
+    | Leaf value -> f value
```

Comment:
Remember the function `f` should take as its first argument something belonging to `tree` this includes the "constructor" i.e. the `Leaf` part not just the arguments to the constructor. In OCaml when pattern-matching you can create an alias to the item you are matching on using `as`. For example here:

```suggestion
    | Leaf _ as v -> f v
```